dbt_semantic_interfaces/parsing/where_filter/where_filter_entity.py

```diff
@@ -27,6 +33,20 @@ def _implements_protocol(self) -> QueryInterfaceEntity:
         return self
 
 
+class MetricStub(ProtocolHint[QueryInterfaceMetric]):
+    """A Metric implementation that just satisfies the protocol.
+
+    QueryInterfaceMetric currently has no methods and the parameter set is created in the factory.
+    """
+
+    @override
+    def _implements_protocol(self) -> QueryInterfaceMetric:
+        return self
+
+    def descending(self, _is_descending: bool) -> QueryInterfaceMetric:  # noqa: D
+        raise InvalidQuerySyntax("descending is invalid in the where parameter and filter spec")
```

Comment:
Ok, I see why this has to be here. We really need to carve out some time to clean up the underlying protocol specs so we don't have these runtime exceptions floating around all over the place.

Reply:
Definitely agree!